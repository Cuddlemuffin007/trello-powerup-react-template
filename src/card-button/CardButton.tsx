import React from "react";

// import { useProvidedTrello } from "../hooks";

export const CardButton = (): JSX.Element => {
    // const t = useProvidedTrello();
    return (
        <div>
            <p>Card Button IFrame content</p>
        </div>
    );
};
