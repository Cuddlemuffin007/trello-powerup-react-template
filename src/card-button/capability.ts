import { Trello, CapabilityProps } from "../@types";

export const getCardButtons = (_t: Trello.PowerUp.IFrame, props: CapabilityProps): Trello.PowerUp.CardButton[] => {
    return [
        {
            icon: props.baseUrl + props.icon.dark,
            text: "Update Candidate Status",
            callback: (tc: Trello.PowerUp.IFrame) =>
                tc.popup({
                    title: "Update Candidate Status",
                    url: "./card-button.html",
                    height: 300,
                }),
        },
    ];
};
