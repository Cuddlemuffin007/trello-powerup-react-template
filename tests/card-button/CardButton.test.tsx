import React from "react";
import { shallow } from "enzyme";

import { CardButton } from "../../src/card-button/CardButton";

describe("CardButton component", () => {
    it("should render the CardButton component", () => {
        const app = shallow(<CardButton />);
        expect(app).toMatchSnapshot();
    });
});
