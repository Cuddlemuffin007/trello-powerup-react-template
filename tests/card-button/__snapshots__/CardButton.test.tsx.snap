// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardButton component should render the CardButton component 1`] = `
<div>
  <p>
    Card Button IFrame content
  </p>
</div>
`;
